src/build-remote/build-remote.cc

```diff
@@ -130,6 +130,38 @@ static int main_build_remote(int argc, char * * argv)
                 for (auto & m : machines) {
                     debug("considering building on remote machine '%s'", m.storeUri);
 
+                    if (std::find(m.systemTypes.begin(), m.systemTypes.end(), "*") != m.systemTypes.end() ||
+                        std::find(m.supportedFeatures.begin(), m.supportedFeatures.end(), "*") != m.supportedFeatures.end()) {
+                        // wildcard is used, so we need to ask the machine what it supports
+                        try {
+                            Activity act(*logger, lvlTalkative, actUnknown, fmt("connecting to '%s'", m.storeUri));
+
+                            auto store = m.openStore();
+
+                            if (hasPrefix(m.storeUri, "ssh://"))
+                                throw Error("Cannot use wildcards with legacy ssh store. Instead use the ssh-ng:// protocol.");
+
+                            store->connect();
+
+                            if (std::find(m.supportedFeatures.begin(), m.supportedFeatures.end(), "*") != m.supportedFeatures.end()) {
+                                m.supportedFeatures.erase("*");
+                                m.supportedFeatures.insert(store->systemFeatures.get().begin(), store->systemFeatures.get().end());
+                            }
```

Comment:
I find it a bit nicer to do that in one pass by keeping an iterator to the found value (also, we can preserve the order when expanding the wildcard rather than adding the new elements at the begining):

```suggestion
                            if (auto wildcardIter = m.supportedFeatures.find("*"); wildcardIter != m.supportedFeatures.end()) {
                                m.supportedFeatures.insert(wildcardIter, store->systemFeatures.get().begin(), store->systemFeatures.get().end());
                                m.supportedFeatures.erase(wildcardIter);
                            }
```